### PR TITLE
[qt-advanced-docking-system] Update to 4.0.3

### LINKED ISSUE #31349: [qt-advanced-docking-system] update to 3.8.4
### Library name

Qt-Advanced-Docking-System

### New version number

3.8.4

### Other information that may be useful (release notes, etc...)

The current version has link failures on Linux because it misses a "target_link_libraries(... xcb)". That was fixed by 3.8.4.

### PR DESCRIPTION
**Pull request by LilyWangLL**
Fixes #31349

Usage test pass on `x64-windows`.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/qt-advanced-docking-system/portfile.cmake [hunk 1]
```diff
@@ -23,19 +21,21 @@ vcpkg_cmake_configure(
...
 vcpkg_cmake_install()
-vcpkg_cmake_config_fixup(PACKAGE_NAME "qtadvanceddocking" CONFIG_PATH "lib/cmake/qtadvanceddocking")
+vcpkg_cmake_config_fixup(PACKAGE_NAME "qt6advanceddocking" CONFIG_PATH "lib/cmake/qt6advanceddocking")
 
+vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/share/qt6advanceddocking/qt6advanceddockingConfig.cmake"
+"include(CMakeFindDependencyMacro)"
+[[include(CMakeFindDependencyMacro)
+find_dependency(Qt6Core)
+find_dependency(Qt6Gui)
+find_dependency(Qt6Widgets)]])
```
Comment (Neumann-A) [inline ports/qt-advanced-docking-system/portfile.cmake:34]:
```suggestion
find_dependency(Qt6 COMPONENTS Core Gui Widgets)]])
```

File: ports/qt-advanced-docking-system/qt.patch [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
-index 653546544..2091bce16 100644
+index 2ec56aa..054ea73 100644
 --- a/src/CMakeLists.txt
 +++ b/src/CMakeLists.txt
 @@ -1,6 +1,5 @@
  cmake_minimum_required(VERSION 3.5)
  project(QtAdvancedDockingSystem LANGUAGES CXX VERSION ${VERSION_SHORT})
 -find_package(QT NAMES Qt6 Qt5 COMPONENTS Core REQUIRED)
- find_package(Qt${QT_VERSION_MAJOR} 5.5 COMPONENTS Core Gui Widgets REQUIRED)
+ find_package(Qt${QT_VERSION_MAJOR} COMPONENTS Core Gui Widgets REQUIRED)
```
Comment (dg0yt) [inline ports/qt-advanced-docking-system/qt.patch:9]: Hardcode `${QT_VERSION_MAJOR}` as `6`. 
- This is the major version declared in the dependencies.
- This is hardcoded in cmake export patching.
- It complements the removal of `-DCMAKE_DISABLE_FIND_PACKAGE_Qt5=TRUE`.
-  `QT_VERSION_MAJOR` is undefined without `find_package(QT)`.

File: ports/qt-advanced-docking-system/qt.patch [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
  if (UNIX AND NOT APPLE)
      include_directories(${Qt${QT_VERSION_MAJOR}Gui_PRIVATE_INCLUDE_DIRS})
```
Comment (dg0yt) [inline ports/qt-advanced-docking-system/qt.patch:11]: Is `QT_VERSION_MAJOR` defined at this point? Are there more uses?

**Review by dg0yt — CHANGES_REQUESTED** [review]

File: ports/qt-advanced-docking-system/qt.patch (hunk 3 above)
Comment (LilyWangLL) [reply ports/qt-advanced-docking-system/qt.patch:11]: It defined in `portfile.cmake`, `-DQT_VERSION_MAJOR=6`.

**Review by jimwang118 — APPROVED** [review]

**Review by dg0yt — APPROVED** [review]
(Overriding my previous "suggest changes".)

**Pull request merged**